image/verify/src/verifier.rs

```diff
@@ -193,6 +199,12 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
             &preamble.vendor_sigs.ecc_sig,
         );
 
+        let vendor_lms_pub_key_idx = preamble.vendor_lms_pub_key_idx;
+        let vendor_lms_info = (
+            &preamble.vendor_pub_keys.lms_pub_keys[vendor_lms_pub_key_idx as usize],
```

Comment:
You need to check that lms_pub_key_idx is in bounds; this can panic.

Reply:
Done.